packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1163,17 +1185,205 @@ class _InactiveElements {
 /// this callback.
 typedef void ElementVisitor(Element element);
 
+/// A handle to the location of a widget in the widget tree.
+///
+/// This class presents a set of methods that can be used from
+/// [StatelessWidget.build] methods and from methods on [State] objects.
+///
+/// [BuildContext] objects are passed to [WidgetBuilder] functions (such as
+/// [StatelessWidget.build]), and are available from the [State.context] member.
+/// Some static functions (e.g. [showDialog], [Theme.of], and so forth) also
+/// take build contexts so that they can act on behalf of the calling widget, or
+/// obtain data specifically for the given context.
+///
+/// Each widget has its own [BuildContext]. In particular, this means that
```

Comment:
Maybe you should explain that the build method's BuildContext will become the ancestor of all of the widgets it creates.  That would help explain why Theme.of(X) fails to find a descendant of X.

I also wish that we weren't coy about the Element tree and the fact that BuildContext is a node in the Element tree which has-a Widget.